チーム紹介編集画面を新規作成

toshinori-m opened the pull request:
### 概要
チーム紹介編集画面（TeamProfileEditPage.tsx）を新規で実装しました。
### 変更内容
- 競技名、種目、メールアドレス、都道府県、チーム名、活動地域、性別、対象年齢、活動実績、その他の編集フォーム
- アイコン画像のアップロード機能
- 入力値に対するバリデーション（例：名前2〜100文字、自己紹介1〜255文字）
- 文字数制限に応じた残り文字数のリアルタイム表示
- API通信失敗時のエラーメッセージ表示

close https://github.com/toshinori-m/stay_connect/issues/218

toshinori-m commented:
まずは表示側の実装部分のみレビューをお願いします。編集・保存処理は次のコミットで追加予定です。

toshinori-m committed:
feat: チーム紹介編集画面の表示部分を作成

hirosi1900day commented on frontend-react/src/pages/teamPages/TeamProfileEditPage.tsx:
```diff
@@ -0,0 +1,326 @@
+    if (!teamId) return
+
+    fetchTeamData(teamId)
+    .then(({ teamData, sportsDisciplineIds, targetAgeIds }) => {
```
みにくいのでインデントを下げたい
```suggestion
      .then(({ teamData, sportsDisciplineIds, targetAgeIds }) => {
```

hirosi1900day commented on frontend-react/src/pages/teamPages/TeamProfileEditPage.tsx:
```diff
@@ -0,0 +1,326 @@
+    sex: "",
+    trackRecord: "",
+    otherBody: ""
+  })
```
stateが大きくなるとコードが読みにくくなる and 不要な再レンダリングが走りパフォーマンスが良くないのでこの大きさになったらstateを個別に分けた方が良いと思います(いつ指摘しようか迷っていた部分)

(updateFormStateいった関数が減らせるまた不要なuseEffectも減らせるかも)

他のコンポーネントで同様な箇所を修正する必要まではないです！

toshinori-m committed:
refactor: フォームのstateを個別に分割し可読性とパフォーマンスを改善

hirosi1900day commented on frontend-react/src/pages/teamPages/TeamProfileEditPage.tsx:
```diff
@@ -0,0 +1,306 @@
+    if (targetAgeIds && targetAgeIds.length > 0) {
+      setTargetAgeSelected(targetAgeIds.map(id => id.toString()))
+    }
+  }
```
ラップ関数は増えれば増えるほど頭のメモリーを消費しながら読む必要が出てくるのでできるだけ使わなくて済むのであれば使わないほうが良いです。
useEffect内でたとえば以下のようなイメージでstate setを行えば関数が不要になる気がしますがいかがでしょうか？
```tsx
setSportsTypeSelected(teamData.sports_type_id?.toString() ?? "")
setTargetAgeSelected((targetAgeIds || []).map(String))
```

hirosi1900day commented on frontend-react/src/pages/teamPages/TeamProfileEditPage.tsx:
```diff
@@ -0,0 +1,306 @@
+    setPendingSportsDisciplineIds(null) // セット後クリア
+  }, [pendingSportsDisciplineIds, sportsDisciplines])
+
+  const handleMultiSelectChange = (
```
関数名見ただけでは役割が何も伝わらない命名なので
handleSelectedValuesChangeのようにもっと直感的な名前にした方が良さそう

toshinori-m committed:
Refactor: 不要なラップ関数を削除し、useEffect内で直接stateを更新するよう修正

toshinori-m commented:
表示側の実装が一通り完了完了出来たかと思っています
次のコミットから、編集・保存処理の実装を行う予定なのでよろしくお願いします！

toshinori-m committed:
feat: チーム紹介編集ページに更新・削除処理を実装

- useActionState を用いて更新処理を実装
- バリデーション追加（チーム名、性別、活動地域など）
- 削除ボタンを追加し、apiClient 経由で削除処理を実行
- 編集完了または削除後、チーム一覧ページに遷移

toshinori-m committed:
Merge branch 'master' into issues/218
